core/client/app/mixins/ed-editor-api.js

```diff
@@ -129,7 +129,7 @@ var EditorAPI = Ember.Mixin.create({
 
         $textarea.focus();
         // Tell the editor it has changed, as programmatic replacements won't trigger this automatically
-        this.sendAction('onChange');
+        this.set('value', $textarea.val());
```

Comment:
I wasn't wanting to define an action, I was wanting to do exactly the same thing @Rovak is having trouble with now, which is fix the events that aren't firing correctly.

Something somewhere is missing, such that the preview doesn't get re-rendered in some browsers (namely firefox) when making changes to the textarea by replacing selections. `sendAction('onChange')` was me shooting in the dark trying to fix it - obviously I thought I had but that's nonsense.

I guess that `this.$().trigger('change')` is the correct incantation over and above `this.set('value', $textarea.val());` ?